src/assets/js/promo.js

```diff
@@ -0,0 +1,52 @@
+import dayjs from 'dayjs';
+import isSameOrBefore from 'dayjs/plugin/isSameOrBefore';
+import isSameOrAfter from 'dayjs/plugin/isSameOrAfter';
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+
+dayjs.extend(isSameOrBefore);
+dayjs.extend(isSameOrAfter);
+dayjs.extend(customParseFormat);
+
+export default () => {
+  return {
+    startDate: '',
+    endDate: '',
+    now: dayjs(),
+    isPromoActive: false,
+
+    async init(startDate, endDate) {
```

Comment:
not sure why this and `setPromoDates` is async.

Reply:
This stops it from firing twice on page load.